src/main/java/elon/Parser.java

```diff
@@ -0,0 +1,120 @@
+    }
+    public void Parse() throws ElonException{
+        while (!inputArr[0].equals("bye")) {
+            if (inputArr[0].equals("list")) {
+                ui.listTasks(list);
+                inputArr = this.nextInput(scanner);
+                continue;
+            } else if (inputArr[0].equals("mark")) {
+                int index = Integer.parseInt(inputArr[1]) - 1;
+                ui.markTask(index, list);
+                try {
+                    storage.saveFile(list);
+                } catch (IOException e) {
+                    System.out.println(e);
+                }
+                inputArr = this.nextInput(scanner);
+                continue;
+            } else if (inputArr[0].equals("unmark")) {
+                int index = Integer.parseInt(inputArr[1]) - 1;
+                ui.unmarkTask(index, list);
+                try {
+                    storage.saveFile(list);
+                } catch (IOException e) {
+                    System.out.println(e);
+                }
+                inputArr = this.nextInput(scanner);
+                continue;
+            } else if (inputArr[0].equals("delete")) {
+                int index = Integer.parseInt(inputArr[1]) - 1;
+                ui.deleteTask(index, list);
+                try {
+                    storage.saveFile(list);
+                } catch (IOException e) {
+                    System.out.println(e);
+                }
+                inputArr = this.nextInput(scanner);
+                continue;
+            } else if (inputArr[0].equals("find")) {
+                String keyword = String.join(" ", Arrays.copyOfRange(inputArr, 1, inputArr.length));
+                ArrayList<Task> matchingTasks = list.findTasks(keyword);
+                ui.showMatchingTasks(matchingTasks);
+                inputArr = this.nextInput(scanner);
+                continue;
+            } else {
+                if (inputArr[0].equals("todo")) {
+                    try {
+                        ui.addToDo(inputArr, list);
+                        storage.saveFile(list);
+                    } catch (ElonException | IOException e) {
+                        System.out.println(e);
+                    }
+                } else if (inputArr[0].equals("deadline")) {
+                    try {
+                        ui.addDeadline(inputArr, list);
+                        storage.saveFile(list);
+                    } catch (ElonException | IOException e) {
+                        System.out.println(e);
+                    }
+                } else if (inputArr[0].equals("event")) {
+                    try {
+                        ui.addEvent(inputArr, list);
+                        storage.saveFile(list);
+                    } catch (ElonException | IOException e) {
+                        System.out.println(e);
+                    }
+                } else {
+                    throw new ElonException("Error. Invalid command.");
+                }
```

Comment:
Have you considered using a switch statement? I don't think a switch statement is compulsory, but it makes readability a lot easier. But nice work for accounting for all the cases!